connection.go

```diff
@@ -147,6 +150,10 @@ func (c *Connection) Connect() error {
 
 	c.run()
 
+	if c.Opts.ConnectionEstablishedHandler != nil {
+		go c.Opts.ConnectionEstablishedHandler(c)
```

Comment:
Are we trying to set k/v pairs on a connection? I'd think that online/offline would be derived from the underlying TCP connection or app usage - and not rely on a k/v.

Reply:
Yeah, from the high-level point of you are right. What I did here is domain-specific thing - the status of the connection on the application level.

Here is the issue I'm trying to solve with this:
* pool establishes the connections
* when a client wants to get a connection from the pool, it will get a live connection (established)
* but a live connection can be `offline` or `online` or whatever

I want a client to be able to get only specific connections from the pool.
I also want clients to not re-implement statuses on the app level. 

Our `Connection` is not a pure network connection. It's already an app connection IMHO :)

Reply:
From your initial question, I think we may just add the `Get` and `Set` methods to set arbitrary key/value for the connection. Then you can set `status` or something else. If you need vs. only fixed field `Status`

Pool doesn't know if the connection is `online` or `offline`. Each network has own logic / codes for network management messages + sign in / sing on. Let's say in a health check for the web service (k8s) we want to return 4xx if no online connections. We can get all connections from the pool and check value of `Status` for each of them. Otherwise, we will have to keep something like a map with `status -> connection` inside (each) gateway and manage it when Pool removes/adds new connections... 

Pool knows if a connection is established or closed - and that's it.

Reply:
Get/Set sounds good to me. I like the idea of setting brand-level / specific metadata.

Reply:
@adamdecaf I've updated the PR by replacing `status` with `kv` to store brand specific key/values (strings). Please, review/approve if everything is ok. It will unblock me with Pool PR. Thanks!